packages/contracts-bedrock/contracts/L1/L2OutputOracleV2.sol

```diff
@@ -0,0 +1,331 @@
+     */
+    constructor(
+        uint256 _l2BlockTime,
+        uint256 _startingBlockNumber,
+        uint256 _startingTimestamp,
+        address _challenger,
+        uint256 _finalizationPeriodSeconds,
+        uint256 _minimumOutputProposalCost
+    ) Semver(2, 0, 0) {
+        require(_l2BlockTime > 0, "L2OutputOracleV2: L2 block time must be greater than 0");
+
+        L2_BLOCK_TIME = _l2BlockTime;
+        CHALLENGER = _challenger;
+        FINALIZATION_PERIOD_SECONDS = _finalizationPeriodSeconds;
+        MINIMUM_OUTPUT_PROPOSAL_COST = _minimumOutputProposalCost;
+
+        initialize(_startingBlockNumber, _startingTimestamp);
+    }
+
+    /**
+     * @notice Initializer.
+     *
+     * @param _startingBlockNumber Block number for the first recoded L2 block.
+     * @param _startingTimestamp   Timestamp for the first recoded L2 block.
+     */
+    function initialize(uint256 _startingBlockNumber, uint256 _startingTimestamp)
+        public
+        initializer
+    {
+        require(
+            _startingTimestamp <= block.timestamp,
+            "L2OutputOracleV2: starting L2 timestamp must be less than current time"
+        );
+
+        startingTimestamp = _startingTimestamp;
+        startingBlockNumber = _startingBlockNumber;
+    }
+
+    /**
+     * @notice Deletes an output proposal based on the given L2 block number.
+     *
+     * @param _l2BlockNumber The L2 block number of the output to delete.
+     */
+    // solhint-disable-next-line ordering
+    function deleteL2Output(uint256 _l2BlockNumber) external {
+        require(
+            msg.sender == CHALLENGER,
+            "L2OutputOracleV2: only the challenger address can delete an output"
+        );
+
+        // Do not allow deleting any outputs that have already been finalized.
+        require(
+            block.timestamp - l2Outputs[_l2BlockNumber].timestamp < FINALIZATION_PERIOD_SECONDS,
+            "L2OutputOracleV2: cannot delete outputs that have already been finalized"
+        );
+
+        // Join the linked list
+        uint256 nextL2Block = nextBlockNumber[_l2BlockNumber];
+        uint256 prevL2Block = previousBlockNumber[_l2BlockNumber];
+        previousBlockNumber[nextL2Block] = prevL2Block;
+        nextBlockNumber[prevL2Block] = nextL2Block;
+        delete nextBlockNumber[_l2BlockNumber];
+        delete previousBlockNumber[_l2BlockNumber];
+
+        // Retrieve the previous high L2 block number
+        if (_l2BlockNumber == highestL2BlockNumber) {
+            highestL2BlockNumber = prevL2Block;
+        }
+
+        // Remove the associated output state
+        delete l2Outputs[_l2BlockNumber];
+        delete bonds[_l2BlockNumber];
+
+        // Transfer the bond to the challenger
+        bool success = SafeCall.call(CHALLENGER, gasleft(), MINIMUM_OUTPUT_PROPOSAL_COST, hex"");
+        require(success, "L2OutputOracleV2: ETH bond transfer failed");
+
+        emit OutputDeleted(_l2BlockNumber);
```

Comment:
For backwards compatibility we might need to keep this consistent with the old event:

https://github.com/ethereum-optimism/optimism/blob/23d3dc792d65432182e4146b896c7bf7f22a6a47/packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol#L166